federation.go

```diff
@@ -24,19 +24,83 @@ type FederationDefinition struct {
 
 // Represents a configured Federation upstream.
 type FederationUpstream struct {
```

Comment:
The tuple of `(vhost, component, name, and value)` is a runtime parameter. An upstream is a specific type (component) of a parameter. So this has to be renamed, e.g. to `RuntimeParameter`.